deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/AgentJanitor.java

```diff
@@ -49,18 +50,23 @@ public AgentJanitor(ServiceContext serviceContext, int minStaleHostThreshold,
         this.maxLaunchLatencyThreshold = maxLaunchLatencyThreshold * 1000;
     }
 
-    private void processStaleHosts(Collection<String> staleHostIds, boolean isMarkedUnreachable) throws Exception {
-        if (staleHostIds.isEmpty()) {
-            return;
-        }
-
+    private Collection<String> getTerminatedHostsFromSource(Collection<String> staleHostIds) throws Exception {
         Collection<String> terminatedHosts = new ArrayList<>(staleHostIds);
         for (String hostId : staleHostIds) {
             Collection<String> resultIds = rodimusManager.getTerminatedHosts(Collections.singletonList(hostId));
             if (resultIds.isEmpty()) {
                 terminatedHosts.remove(hostId);
             }
         }
+        return terminatedHosts;
+    }
+
+    private void processStaleHosts(Collection<String> staleHostIds, boolean isMarkedUnreachable) throws Exception {
+        if (staleHostIds.isEmpty()) {
```

Comment:
This collection can be nullable? if yes I guess you could check using Collections